framework/auth/core.py

```diff
@@ -1331,8 +1334,63 @@ def n_projects_in_common(self, other_user):
         """Returns number of "shared projects" (projects that both users are contributors for)"""
         return len(self.get_projects_in_common(other_user, primary_keys=True))
 
+    def has_inst_auth(self, inst):
+        if inst in self.affiliated_institutions:
+            return True
+        return False
+
+    affiliated_institutions = fields.ForeignField('institution', list=True)
+
 
 def _merge_into_reversed(*iterables):
     '''Merge multiple sorted inputs into a single output in reverse order.
     '''
     return sorted(itertools.chain(*iterables), reverse=True)
+
+
+class Institution(StoredObject):
+
+    _id = fields.StringField(index=True, unique=True, primary=True)
+    name = fields.StringField(required=True)
+    logo_name = fields.StringField()
```

Comment:
Should this be required? It looks like you're counting on the existence of a logo in the front end.